### PR TITLE
Semantic Search API Gated Public Preview

### PR DESCRIPTION
**Pull request by shmed**
This is a continuation of this PR: https://github.com/Azure/azure-rest-api-specs/pull/13487

Yanoosh is temporarily out so address the remaining comments in this PR.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by shmed** [comment]
@jhendrixMSFT \ @AlexGhiondea 
This PR is a continuation of Janusz's PR. Since he is OOF right now, I created this new branch that resolve some of the remaining comments (notably the speller and prettier failures)

**Comment by tjprescott** [comment]
@shmed please provide the breaking change justification mentioned in the comments above.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 1]
```diff
@@ -245,6 +251,93 @@
...
+              "values": [
+                {
+                  "value": "none",
+                  "name": "None",
+                  "description": "Query language not specified."
+                },
+                {
+                  "value": "en-us",
+                  "name": "EnUs",
+                  "description": "English"
```
Comment (heaths) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:274]: I would use "US English" or "English (US)" as en-us is locale-specific to the US.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 2]
```diff
@@ -245,6 +251,93 @@
...
+                  "description": "Do not return answers for the query."
+                },
+                {
+                  "value": "extractive",
+                  "name": "Extractive",
+                  "description": "Extracts answer candidates from the contents of the documents returned in response to a query expressed as a question in natural language."
+                }
+              ]
+            },
+            "description": "This parameter is only valid if the query type is 'semantic'. If set, the query returns answers extracted from key passages in the highest ranked documents. The number of answers returned can be configured by appending the |count-<number of answers> option after the answers parameter value. Default is 1.",
```
Comment (heaths) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:336]: What the pipe "|" character for before "count"?

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 3]
```diff
@@ -992,6 +1085,54 @@
       "additionalProperties": true,
       "description": "A single bucket of a facet query result. Reports the number of documents with a field value falling within a particular range or having a particular value or interval."
     },
+    "AnswerResult": {
+      "properties": {
+        "score": {
+          "type": "integer",
+          "format": "int64",
```
Comment (heaths) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1092]: Isn't the score using a double? If this really is an integer and, if inconsistent as I suspect, can't change, then what is the value range? 0 to 100? That should be documented.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 4]
```diff
@@ -992,6 +1085,54 @@
...
+          "readOnly": true,
+          "description": "The score value represents how relevant the answer is to the the query relative to other answers returned for the query."
+        },
+        "key": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The key of the document the answer was extracted from.",
+          "x-ms-client-name": "Key"
```
Comment (heaths) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1100]: 1. You don't have this on `score`.
2. You don't really need it when it's the same as the property. Our generators will case as appropriate for the language.

Comment (heaths) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1100]: For 2, this applies everywhere else in this PR as well.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 5]
```diff
@@ -1057,6 +1210,14 @@
           "x-nullable": false,
           "description": "The relevance score of the document compared to other documents returned by the query."
         },
+        "@search.rerankerScore": {
+          "type": "number",
+          "format": "double",
```
Comment (heaths) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1215]: In regard to comment above, this is a double. Why isn't the other score a double?

**Review by heaths — APPROVED** [review]
A few comments/questions, but no reason to hold the PR up if those are addressed as needed.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 2 above)
Comment (shmed) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:336]: This is the syntax Azure Search currently use to add "sub" parameters at query time.  An example input would be
answers=extractive|count-1

We already use a similar syntax for highlights:
highlight=title|count-5,description|count-3

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 5 above)
Comment (shmed) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1215]: they should all be double

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 6]
```diff
@@ -1230,10 +1404,87 @@
...
+            "description": "Do not return answers for the query."
+          },
+          {
+            "value": "extractive",
+            "name": "Extractive",
+            "description": "Extracts answer candidates from the contents of the documents returned in response to a query expressed as a question in natural language."
+          }
+        ]
+      },
+      "description": "This parameter is only valid if the query type is 'semantic'. If set, the query returns answers extracted from key passages in the highest ranked documents. The number of answers returned can be configured by appending the |count-<number of answers> option after the answers parameter value. Default is 1."
```
[the remark this answers is not in the document]
Comment (shmed) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1463]: Same as my comment above. The pipe is part of the current syntax AZS use.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 2 above)
Comment (heaths) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:336]: Thanks for explaining. I would suggest rewording to make that clearer for customers who may not know. I help develop the Search SDK and wasn't aware. 🙂

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 7]
```diff
@@ -271,7 +271,7 @@
                 {
                   "value": "en-us",
                   "name": "EnUs",
-                  "description": "English"
+                  "description": "English (US)"
```
Comment (heaths) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:274]: I probably should've brought this up before, but it just occurred to me now: other language references in the swagger (all of them, if memory serves me correctly) are just the language. Only a few languages have culture variations. So is this really "en-us", or should it just be "en"? Does culture (number formatting, etc.) really play a factor here?

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 8]
```diff
@@ -992,6 +1085,54 @@
       "additionalProperties": true,
       "description": "A single bucket of a facet query result. Reports the number of documents with a field value falling within a particular range or having a particular value or interval."
     },
+    "AnswerResult": {
+      "properties": {
```
[the remark this answers is not in the document]
Comment (heaths) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1089]: I take it this change wasn't made because none of these should ever return an explicit `null`?

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json [hunk 9]
```diff
@@ -1240,6 +1236,7 @@
           },
           "readOnly": true,
           "x-ms-client-name": "Captions",
+          "x-nullable": true,
```
Comment (heaths) [inline specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1239]: Note: I see this was added, so maybe that answers my question I re-activated, i.e. perhaps all those properties would not return an explicit `null`.

**Review by heaths — APPROVED** [review]

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 7 above)
Comment (shmed) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:274]: I believe it does. We will be adding various speller lexicons, many of them being culture-specific.

File: specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (shmed) [reply specification/search/data-plane/Azure.Search/preview/2020-06-30-Preview/searchindex.json:1089]: I missed them. The "highlights" are nullable, the others are not

**Review by JeffreyRichter — APPROVED** [review]
